src/test/java/com/juniorstart/juniorstart/service/UserProfileIntegrationTest.java

```diff
@@ -0,0 +1,131 @@
+package com.juniorstart.juniorstart.service;
+
+import com.juniorstart.juniorstart.exception.BadRequestException;
+import com.juniorstart.juniorstart.model.*;
+import com.juniorstart.juniorstart.payload.PagedResponse;
+import com.juniorstart.juniorstart.payload.UserRoleOrTechnologyRequest;
+import com.juniorstart.juniorstart.repository.UserDao;
+import com.juniorstart.juniorstart.repository.UserProfileRepository;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.test.context.junit4.SpringRunner;
+import java.util.*;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+
+/** Represents an user service.
+ * @author Rafał Maduzia
+ * @version 1.0
+ */
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@AutoConfigureTestDatabase(replace= AutoConfigureTestDatabase.Replace.NONE)
+public class UserProfileIntegrationTest {
+
+    @SpyBean
+    private UserProfileService userProfileService;
+
+    @Autowired
+    private UserDao userRepository;
+
+    @Autowired
+    private UserProfileRepository userProfileRepository;
+
+    private List<String> technologyList  = new ArrayList<>();
+    private List<UserRole> userRoleList  = new ArrayList<>();
+    private List<String> userRoleListThrowException  = new ArrayList<>();
+    private int page = 0;
+    private int size = 8;
+
+    @Before
+    public void setUp() {
+        User user;
+        user = User.builder()
+                .name("Test")
+                .email("test2@test.com")
+                .emailVerified(true)
+                .password("Test%123")
+                .provider(AuthProvider.google).build();
+        userRepository.save(user);
+
+        UserTechnology userTechnology = new UserTechnology();
+        userTechnology.setTechnologyName("Java");
+
+        UserProfile userProfile = new UserProfile();
+        userProfile.setUser(user);
+        userProfile.setUserRole(UserRole.MENTOR);
+        userProfile.addUserTechnology(userTechnology);
+        userProfileRepository.save(userProfile);
+    }
+
+    @Test
+    public void should_FindUserProfileByRoleAndTechnology() {
+        technologyList.add("java");
+        userRoleList.add(UserRole.MENTOR);
+        PagedResponse<UserProfile> foundUser = userProfileService.findByTechnologyAndRole(technologyList, userRoleList, page, size);
+        assertNotEquals(new ArrayList<>(Collections.emptyList()), foundUser.getContent());
```

Comment:
Nie myślisz, że lepiej będzie przetestować w ten sposób: 
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
@ExtendWith(MockitoExtension.class)
public class ProjectServiceTest {


    ProjectRepository projectRepository = Mockito.mock(ProjectRepository.class);
    UserDao userDao = Mockito.mock(UserDao.class);
    ProjectLikeRepository projectLikeRepository = Mockito.mock(ProjectLikeRepository.class);
    TechnologyRepository technologyRepository = Mockito.mock(TechnologyRepository.class);

    ProjectService projectService = new ProjectService(projectRepository, userDao, projectLikeRepository, technologyRepository);

    Instant createdAt;
    Project project;
    Technology technology;
    ProjectTodo projectTodo;
    UserSummary userSummary;
    User user;
    UserSummary mentor;
    ProjectComment projectComment;
    Pageable pageable;
    List<Project> projectList;
    List<Long> listOfLikes;
    int total;
    int start;
    int end;
    List<Project> output;
    List<ProjectTodo> projectTodos = new ArrayList<>();
    PageImpl page;
    List<User> users;
    UserPrincipal userPrincipal;
    List<ProjectComment> projectComments = new ArrayList<>();
    ProjectRequest projectRequest;
    ProjectLike projectLike;

    @Before
    public void initialize() throws Exception {
        technology = Technology.builder().technologyType(TechnologyType.FRONTEND).description("Some description").id(1l).title("Some title").build();
        projectTodo = ProjectTodo.builder().user(user).title("Some title").project(project).id(1l).build();
        projectComment = ProjectComment.builder().project(project).body("Some body").user(user).id(1l).build();
        createdAt = new SimpleDateFormat("yyyy-MM-dd").parse("2020-12-31").toInstant();
        users = new ArrayList<>();
        user = new User();
        user.setPrivateId(UUID.randomUUID());
        user.setPublicId(21l);
        user.setEmailVerified(true);
        user.setName("Some name");
        user.setProvider(AuthProvider.local);
        user.setPassword("Somepassword");
        user.setAge(12);
        user.setHiddenFromSearch(true);
        user.setImageUrl("SomeImageUrl");
        user.setUserStatus(UserStatus.LOOKING_FOR_A_JOB);
        user.setEmail("some@example.com");
        user.setCreatedAt(createdAt);
        user.setUpdatedAt(createdAt);
        project = new Project();
        project.setBody("Some body");
        project.setDescription("Some description");
        project.setNumberOfSeats(12l);
        project.setTitle("Some title");
        project.setCreatedAt(createdAt);
        project.setCreatedBy(user.getPublicId());
        project.setCreatedAt(createdAt);
        project.setUpdatedBy(user.getPublicId());
        project.setId(12l);

        project.setName("Some Easy name");
        pageable = PageRequest.of(0, 10, Sort.Direction.DESC, "createdAt");
        projectList = new ArrayList<>();
        projectList.add(project);
        page = MockUtil.createMockPage(projectList);

        users.add(user);
        projectTodo.setUser(user);
        projectTodos.add(projectTodo);
        project.setProjectTodos(projectTodos);
        project.setProjectComments(projectComments);
        listOfLikes = new ArrayList<>();
        listOfLikes.add((long) 53412);
        listOfLikes.add((long) 1323232);
        listOfLikes.add((long) 13434);
        userPrincipal = UserPrincipal.create(user);
        projectRequest = new ProjectRequest();
        projectRequest.setBody(project.getBody());
        projectRequest.setDescription(project.getDescription());
        projectRequest.setName(project.getName());
        projectRequest.setNumberOfSeats(project.getNumberOfSeats());
        projectRequest.setRepository(project.getRepository());
        projectRequest.setTitle(project.getTitle());
        projectLike = new ProjectLike();
        projectLike.setUser(user);
        projectLike.setProject(project);
    }